packages/web/src/layouts/pool/pool-detail/components/staking/staking-content/incentivized-token-detail-tooltip-content/IncentivizeTokenDetailTooltipContent.tsx

```diff
@@ -19,65 +19,71 @@ const IncentivizeTokenDetailTooltipContent: React.FC<Props> = ({ poolStakings }:
   const { getGnotPath } = useGnotToGnot();
   const { t } = useTranslation();
 
-  const displayRemainingAmount = useCallback((staking: PoolStakingModel) => {
+  const displayRemainingAmount = (staking: PoolStakingModel) => {
     return staking.incentiveType !== "INTERNAL";
-  }, []);
+  };
 
-  const incentiveTypeText = useCallback((type: INCENTIVE_TYPE) => {
+  const incentiveTypeText = (type: INCENTIVE_TYPE) => {
     return capitalize(type);
-  }, []);
+  };
+
+  const sortedStakings = React.useMemo(() => {
```

Comment:
It seems like it would be good to add guards since `NaN` could occur if `startTimestamp` is an invalid string, and `poolStaking` could potentially be `undefined`. 🤔